### PR TITLE
Update e2e tests for unconfigured codemodules scenarios

### PR DESCRIPTION
**Pull request by andriisoldatenko**
## Description

Please include the following:

This PR updates e2e test with new check.

## How can this be tested?

```
make test/e2e/cloudnative
make test/e2e/applicationmonitoring
```

## Checklist

- [x] Unit tests have been updated/added
- [x] PR is labeled accordingly with a single label
- [x] I have read and understood the [contribution guidelines](https://github.com/Dynatrace/dynatrace-operator/blob/main/CONTRIBUTING.md)

### CODE REVIEW

File: test/helpers/components/codemodules/codemodules.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	"sigs.k8s.io/e2e-framework/pkg/features"
+)
+
+const (
+	RuxitAgentProcFile     = "ruxitagentproc.conf"
+	DaemonSetCSIdriverName = "dynatrace-oneagent-csi-driver"
+)
+
+func CheckRuxitAgentProcFileHasNoConnInfo(testDynakube dynatracev1beta1.DynaKube) features.Func {
+	return func(ctx context.Context, t *testing.T, e *envconf.Config) context.Context {
```
Comment (waodim) [inline test/helpers/components/codemodules/codemodules.go:28]: very minor nitpick but we (almost?) always name envconf.Config variables with environmentConfig, maybe you want to consider this out of consistency perspective

Comment (andriisoldatenko) [reply test/helpers/components/codemodules/codemodules.go:28]: @waodim yes and no :) 
```
test/scenarios/applicationmonitoring/dataingest.go:	"sigs.k8s.io/e2e-framework/pkg/envconf"
test/scenarios/applicationmonitoring/dataingest.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/scenarios/applicationmonitoring/dataingest.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/scenarios/applicationmonitoring/without_csi.go:	"sigs.k8s.io/e2e-framework/pkg/envconf"
test/scenarios/applicationmonitoring/without_csi.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/scenarios/applicationmonitoring/without_csi.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/scenarios/cloudnative/install.go:	"sigs.k8s.io/e2e-framework/pkg/envconf"
test/scenarios/cloudnative/install.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/scenarios/applicationmonitoring/labelversiondetection.go:	"sigs.k8s.io/e2e-framework/pkg/envconf"
test/scenarios/applicationmonitoring/labelversiondetection.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/scenarios/applicationmonitoring/read_only_csi_volume.go:	"sigs.k8s.io/e2e-framework/pkg/envconf"
test/scenarios/applicationmonitoring/read_only_csi_volume.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/scenarios/applicationmonitoring/read_only_csi_volume.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/scenarios/cloudnative/specific_agent_version/specific_agent_version.go:	"sigs.k8s.io/e2e-framework/pkg/envconf"
test/scenarios/cloudnative/specific_agent_version/specific_agent_version.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/scenarios/cloudnative/proxy/proxy.go:	"sigs.k8s.io/e2e-framework/pkg/envconf"
test/scenarios/cloudnative/proxy/proxy.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/scenarios/cloudnative/proxy/proxy.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/scenarios/cloudnative/network/network_problems.go:	"sigs.k8s.io/e2e-framework/pkg/envconf"
test/scenarios/cloudnative/network/network_problems.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/scenarios/cloudnative/public_registry/public_registry.go:	"sigs.k8s.io/e2e-framework/pkg/envconf"
test/scenarios/cloudnative/public_registry/public_registry.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/scenarios/cloudnative/public_registry/public_registry.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/scenarios/cloudnative/public_registry/public_registry.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/scenarios/cloudnative/codemodules/codemodules.go:	"sigs.k8s.io/e2e-framework/pkg/envconf"
test/scenarios/cloudnative/codemodules/codemodules.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/scenarios/cloudnative/codemodules/codemodules.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/scenarios/cloudnative/codemodules/codemodules.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/scenarios/cloudnative/codemodules/codemodules.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/scenarios/cloudnative/disabled_auto_injection/disabled_auto_injection.go:	"sigs.k8s.io/e2e-framework/pkg/envconf"
test/scenarios/cloudnative/disabled_auto_injection/disabled_auto_injection.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/scenarios/dummy/dummy_test.go:	"sigs.k8s.io/e2e-framework/pkg/envconf"
test/scenarios/dummy/dummy_test.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/scenarios/dummy/dummy_test.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/scenarios/activegate/installation.go:	"sigs.k8s.io/e2e-framework/pkg/envconf"
test/scenarios/activegate/installation.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/scenarios/activegate/installation.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/scenarios/activegate/installation.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/scenarios/activegate/installation.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/scenarios/activegate/installation.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/helpers/logs/logs.go:	"sigs.k8s.io/e2e-framework/pkg/envconf"
test/helpers/logs/logs.go:func ReadLog(ctx context.Context, t *testing.T, environmentConfig *envconf.Config, namespace, podName, containerName string) string { //nolint:revive
test/helpers/components/oneagent/daemonset.go:	"sigs.k8s.io/e2e-framework/pkg/envconf"
test/helpers/components/oneagent/daemonset.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/helpers/components/oneagent/daemonset.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/helpers/kubeobjects/manifests/installation.go:	"sigs.k8s.io/e2e-framework/pkg/envconf"
test/helpers/kubeobjects/manifests/installation.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/helpers/kubeobjects/manifests/installation.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/helpers/kubeobjects/manifests/installation.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/helpers/kubeobjects/manifests/installation.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/helpers/kubeobjects/manifests/installation.go:func installFromSingleUrl(t *testing.T, ctx context.Context, environmentConfig *envconf.Config, yamlUrl string) {
test/helpers/kubeobjects/manifests/installation.go:func uninstallFromSingleUrl(t *testing.T, ctx context.Context, environmentConfig *envconf.Config, yamlUrl string) {
test/helpers/kubeobjects/namespace/namespace.go:	"sigs.k8s.io/e2e-framework/pkg/envconf"
test/helpers/kubeobjects/namespace/namespace.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/helpers/kubeobjects/namespace/namespace.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/helpers/kubeobjects/pod/wait.go:	"sigs.k8s.io/e2e-framework/pkg/envconf"
test/helpers/kubeobjects/pod/wait.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/helpers/kubeobjects/pod/wait.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/helpers/kubeobjects/pod/wait.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/helpers/kubeobjects/deployment/wait.go:	"sigs.k8s.io/e2e-framework/pkg/envconf"
test/helpers/kubeobjects/deployment/wait.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/helpers/kubeobjects/environment/environment.go:	"sigs.k8s.io/e2e-framework/pkg/envconf"
test/helpers/kubeobjects/environment/environment.go:		environment.Setup(envfuncs.CreateKindCluster(envconf.RandomName("operator-test", 10)))
test/helpers/kubeobjects/environment/environment.go:	cfg, _ := envconf.NewFromFlags()
test/helpers/kubeobjects/statefulset/wait.go:	"sigs.k8s.io/e2e-framework/pkg/envconf"
test/helpers/kubeobjects/statefulset/wait.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/helpers/components/operator/installation.go:	"sigs.k8s.io/e2e-framework/pkg/envconf"
test/helpers/components/operator/installation.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/helpers/components/operator/installation.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/helpers/components/operator/installation.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/helpers/kubeobjects/daemonset/wait.go:	"sigs.k8s.io/e2e-framework/pkg/envconf"
test/helpers/kubeobjects/daemonset/wait.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/helpers/components/dynakube/dynakube.go:	"sigs.k8s.io/e2e-framework/pkg/envconf"
test/helpers/components/dynakube/dynakube.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/helpers/components/dynakube/dynakube.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/helpers/components/dynakube/dynakube.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/helpers/components/dynakube/dynakube.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/helpers/components/codemodules/codemodules.go:	"sigs.k8s.io/e2e-framework/pkg/envconf"
test/helpers/components/codemodules/codemodules.go:	return func(ctx context.Context, t *testing.T, e *envconf.Config) context.Context {
test/helpers/components/csi/daemonset.go:	"sigs.k8s.io/e2e-framework/pkg/envconf"
test/helpers/components/csi/daemonset.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/helpers/components/activegate/installation.go:	"sigs.k8s.io/e2e-framework/pkg/envconf"
test/helpers/components/activegate/installation.go:func ReadActiveGateLog(ctx context.Context, t *testing.T, environmentConfig *envconf.Config, testDynakube *dynatracev1beta1.DynaKube, component string) string {
test/helpers/proxy/proxy.go:	"sigs.k8s.io/e2e-framework/pkg/envconf"
test/helpers/proxy/proxy.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/helpers/sampleapps/sampleapp.go:	"sigs.k8s.io/e2e-framework/pkg/envconf"
test/helpers/sampleapps/sampleapp.go:	return func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
test/helpers/sampleapps/sampleapp.go:func (app sampleApp) installSCC(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
test/helpers/sampleapps/sampleapp.go:	return func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
test/helpers/sampleapps/sampleapp.go:func (app sampleApp) uninstallSCC(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
test/helpers/tenant/secrets.go:	"sigs.k8s.io/e2e-framework/pkg/envconf"
test/helpers/tenant/secrets.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/helpers/sampleapps/pod.go:	"sigs.k8s.io/e2e-framework/pkg/envconf"
test/helpers/sampleapps/pod.go:func (app SamplePod) RestartHalf(ctx context.Context, t *testing.T, config *envconf.Config) context.Context {
test/helpers/sampleapps/pod.go:func (app SamplePod) Restart(ctx context.Context, t *testing.T, config *envconf.Config) context.Context {
test/helpers/sampleapps/pod.go:func (app SamplePod) doRestart(ctx context.Context, t *testing.T, config *envconf.Config, restartFunc restartFunc) context.Context {
test/helpers/istio/install.go:	"sigs.k8s.io/e2e-framework/pkg/envconf"
test/helpers/istio/install.go:func AddIstioNetworkAttachment(namespace corev1.Namespace) func(ctx context.Context, environmentConfig *envconf.Config, t *testing.T) (context.Context, error) {
test/helpers/istio/install.go:	return func(ctx context.Context, environmentConfig *envconf.Config, t *testing.T) (context.Context, error) {
test/helpers/istio/install.go:func AssertIstioNamespace() func(ctx context.Context, environmentConfig *envconf.Config, t *testing.T) (context.Context, error) {
test/helpers/istio/install.go:	return func(ctx context.Context, environmentConfig *envconf.Config, t *testing.T) (context.Context, error) {
test/helpers/istio/install.go:func AssertIstiodDeployment() func(ctx context.Context, environmentConfig *envconf.Config, t *testing.T) (context.Context, error) {
test/helpers/istio/install.go:	return func(ctx context.Context, environmentConfig *envconf.Config, t *testing.T) (context.Context, error) {
test/helpers/istio/install.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/helpers/istio/install.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/helpers/istio/install.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/helpers/istio/install.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/helpers/sampleapps/deployment.go:	"sigs.k8s.io/e2e-framework/pkg/envconf"
test/helpers/sampleapps/deployment.go:func (app SampleDeployment) RestartHalf(ctx context.Context, t *testing.T, config *envconf.Config) context.Context {
test/helpers/sampleapps/deployment.go:func (app SampleDeployment) Restart(ctx context.Context, t *testing.T, config *envconf.Config) context.Context {
test/helpers/sampleapps/deployment.go:func (app SampleDeployment) doRestart(ctx context.Context, t *testing.T, config *envconf.Config, restartFunc restartFunc) context.Context {
test/helpers/sampleapps/curl.go:	"sigs.k8s.io/e2e-framework/pkg/envconf"
test/helpers/sampleapps/curl.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/helpers/sampleapps/curl.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/helpers/sampleapps/curl.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/helpers/sampleapps/curl.go:	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
test/helpers/sampleapps/base/sampleapp.go:	"sigs.k8s.io/e2e-framework/pkg/envconf"
test/helpers/sampleapps/base/sampleapp.go:	Restart(ctx context.Context, t *testing.T, config *envconf.Config) context.Context
test/helpers/sampleapps/base/sampleapp.go:	RestartHalf(ctx context.Context, t *testing.T, config *envconf.Config) context.Context

```

Comment (andriisoldatenko) [reply test/helpers/components/codemodules/codemodules.go:28]: we currently have 3 options:
- environmentConfig, config, c envconfig :)
but I agree to rename all of them into one convention.

File: test/helpers/components/codemodules/codemodules.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+)
+
+const (
+	RuxitAgentProcFile     = "ruxitagentproc.conf"
+	DaemonSetCSIdriverName = "dynatrace-oneagent-csi-driver"
```
[the remark this answers is not in the document]
Comment (andriisoldatenko) [reply test/helpers/components/codemodules/codemodules.go:24]: oh cool, I'll reuse it.

File: test/helpers/components/codemodules/codemodules.go (hunk 1 above)
Comment (waodim) [reply test/helpers/components/codemodules/codemodules.go:28]: okay, then nevermind! :slightly_smiling_face:

Comment (andriisoldatenko) [reply test/helpers/components/codemodules/codemodules.go:28]: i'll create separate PR for this thing.

File: test/helpers/components/codemodules/codemodules.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (andriisoldatenko) [reply test/helpers/components/codemodules/codemodules.go]: fixed in 19dd7579

File: test/helpers/components/codemodules/codemodules.go (hunk 1 above)
Comment (andriisoldatenko) [reply test/helpers/components/codemodules/codemodules.go:28]: @waodim done in https://github.com/Dynatrace/dynatrace-operator/pull/1990

**Review by waodim — APPROVED** [review]
Code LGTM and tests ran through :slightly_smiling_face:

**Pull request merged**